### DISCUSSION TITLE
Distance Trevelled in a Open Field Test

### DISCUSSION THREAD

**Discussion by diegogallo96** [bonsai-rx/bonsai#1250] [Q&A]
Hi bonsai users,
I want to make a workflow that calculates the distance traveled of a rat in an open field. I attach the workflow that I carry out. It works fine, the only problem I have is that I can't rescale correctly, I want the result in centimeters, not in pixels.

So I have 2 problems:
1- I'm not sure if I pass the parameters to the rescale node correctly.
2- When I use 2 test videos where I make the same object travel the same distance, but at different speeds, in the video where the object travels faster it registers a shorter distance (attached the videos so you can check it, they are in mp4 becouse i cant uplouded in avi, you can convert them).

Thank you so much in advance!

![workflow](https://user-images.githubusercontent.com/110508740/218586974-59c67a69-1aad-4bfb-9ad1-1f3b0bae2a90.png)

https://user-images.githubusercontent.com/110508740/218586982-d527f53b-2ff6-40f5-8c3e-0ced79cabb13.mp4

https://user-images.githubusercontent.com/110508740/218586988-91a2f735-2682-4abc-8e9e-429f616d5e80.mp4

**Comment by bruno-f-cruz** [comment]
`Rescale` will just implement a linear interpolation given the properties you pass to it.
E.g. if you define your input max and min values to be 0 and 1, respectively and your outgoing values to be 0, 10, the operator will scale all points by 10x.

Regarding the second question, I am not sure I follow, the videos look identical when it comes to distance traveled, shouldn't the result of the accumulation of velocity be identical across the two videos as you observed?

**Comment by diegogallo96** [comment]
Bruno thanks for the quick response, I think I can know how to continue.

Is there a way to measure how many pixels have an object that is in the film? from that I could make use of the rescale node...

**Reply by bruno-f-cruz** [reply]
`BinaryRegionAnalysis` outputs this information from a set of segmented contours (e.g. using `FindContours`).
